### ISSUE TITLE
With layer-shell enabled tooltip doesn't show up for the right-most module

### ISSUE THREAD

**Issue by eternal-sorrow** [Alexays/Waybar#532]
Maybe it's because of some incorrect positioning? may it be rendered outside of screen boundaries?

**Comment by eternal-sorrow** [comment]
Not just right-most module is affected, but any module close enough to the right edge. My guess is, if the tooltip would be shown partly off-screen, it isn't shown at all.

**Comment by alebastr** [comment]
Confirmed: for modules near the right edge vertical component of tooltip position is not adjusted to fit the screen boundaries (horizontal is adjusted though). I.e. if your bar is at the bottom, tooltips would be rendered outside of the screen.

It could be a bug in sway or in gtk-layer-shell popup positioning code. I'll try debugging gtk-layer-shell when I have time for that.

**Comment by alebastr** [comment]
```
[2338775.340]  -> xdg_wm_base@31.create_positioner(new id xdg_positioner@37)
[2338775.396]  -> xdg_positioner@37.set_size(181, 32)
[2338775.444]  -> xdg_positioner@37.set_anchor_rect(89, 344, 54, 44)
[2338775.520]  -> xdg_positioner@37.set_offset(0, 0)
[2338775.567]  -> xdg_positioner@37.set_anchor(2)
[2338775.605]  -> xdg_positioner@37.set_gravity(2)
[2338775.642]  -> xdg_positioner@37.set_constraint_adjustment(9)
[2338775.673]  -> xdg_wm_base@31.get_xdg_surface(new id xdg_surface@40, wl_surface@39)
[2338775.719]  -> xdg_surface@40.get_popup(new id xdg_popup@35, nil, xdg_positioner@37)
<...>
[2338777.020] xdg_popup@35.configure(-26, 388, 181, 32)
```
Constraint adjustment is `slide_x & flip_y`. Appears that wlroots is slightly buggy with this combination, which is very easy to fix.
I'll send a wlroots PR by the end of week.

**Comment by eternal-sorrow** [comment]
Fixed with wlroots 0.10.1
